scripts/scil_visualize_connectivity.py

```diff
@@ -0,0 +1,179 @@
+                   vmin=0.0, vmax=np.max(matrix))
+
+    if args.write_values:
+        if np.prod(matrix.shape) > 1000:
+            logging.warning('Large matrix, please consider not using '
+                            '--write_values.')
+        ax = write_values(ax, matrix, args.write_values)
+
+    if args.display_legend:
+        fig.colorbar(im, ax=ax)
+
+    if args.name_axis:
+        x_ticks = np.arange(matrix.shape[0])
+        y_ticks = np.arange(matrix.shape[1])
+
+        if args.labels_list:
+            labels_list = np.loadtxt(args.labels_list, dtype=np.int16).tolist()
+
+            if not args.reorder_json and not args.lookup_table:
+                if len(labels_list) != matrix.shape[0] \
+                        or len(labels_list) != matrix.shape[1]:
+                    logging.warning('The provided matrix not the same size as '
+                                    'the labels list.')
+                x_legend = labels_list[0:matrix.shape[0]]
+                y_legend = labels_list[0:matrix.shape[1]]
+
+            if args.reorder_json:
+                filename, key = args.reorder_json
+                with open(filename) as json_data:
+                    config = json.load(json_data)
+
+                    x_legend = config[key][0]
+                    y_legend = config[key][1]
+
+            if args.lookup_table:
+                logging.warning('Using a lookup table, make sure the reordering '
+                                'json contain labels, not coordinates')
+                with open(args.lookup_table) as json_data:
+                    lut = json.load(json_data)
+
+                x_legend = []
+                y_legend = []
+                if args.reorder_json:
+                    x_list = config[key][0]
+                    y_list = config[key][1]
+                else:
+                    x_list = labels_list[0:matrix.shape[0]]
+                    y_list = labels_list[0:matrix.shape[1]]
+
+                x_legend = [lut[str(x)] if str(x) in lut else str(x)
+                            for x in x_list]
+                y_legend = [lut[str(x)] if str(x) in lut else str(x)
+                            for x in y_list]
+
+        else:
+            x_legend = x_ticks
+            y_legend = y_ticks
+
+        if len(x_ticks) != len(x_legend) \
+                or len(y_ticks) != len(y_legend):
+            logging.warning('Legend is not the same size as the data.'
+                            'Make sure you are using the same reordering json.')
+        plt.xticks(x_ticks, x_legend,
+                   rotation=args.axis_text_angle[0],
+                   fontsize=args.axis_text_size[0])
+        plt.yticks(y_ticks, y_legend,
+                   rotation=args.axis_text_angle[1],
+                   fontsize=args.axis_text_size[1])
+
+    if args.show_only:
+        plt.show()
+    else:
+        plt.savefig(args.out_png, dpi=300, bbox_inches='tight')
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
requirements.txt need to include matplotlib please

./bash_cmd.sh
Traceback (most recent call last):
  File "/Users/desm2239/Research/Source//scilpy/scripts/scil_visualize_connectivity.py", line 22, in <module>
    import matplotlib.pyplot as plt
ModuleNotFoundError: No module named 'matplotlib'
Traceback (most recent call last):
  File "/Users/desm2239/Research/Source//scilpy/scripts/scil_visualize_connectivity.py", line 22, in <module>
    import matplotlib.pyplot as plt
ModuleNotFoundError: No module named 'matplotlib'

Reply:
I'm going to add it.

Also for the text placement, does reducing the font works ?
I mean, is the quality high enough to be able to read something much smaller ?

Reply:
This works well on my end. Maybe you could make this the default or have a heuristic depending on the size of the matrix... thanks!